### ISSUE TITLE
Android: Crash on Profile view

### ISSUE THREAD

**Issue by shuvashish76** [samuelclay/NewsBlur#1651]
Steps to produce:
1. Open NewsBlur > Profile (top_left) or Navigation Activities button (top_right)
2. App crashes

Crash logs:

```
FATAL EXCEPTION: main
Process: com.newsblur, PID: 17724
java.lang.NullPointerException: Attempt to invoke interface method 'int java.lang.CharSequence.length()' on a null object reference
	at android.text.SpannableStringBuilder.append(SpannableStringBuilder.java:269)
	at com.newsblur.view.ActivitiesAdapter.getSharedStoryContent(ActivitiesAdapter.java:161)
	at com.newsblur.view.ActivitiesAdapter.getTextForActivity(ActivitiesAdapter.java:56)
	at com.newsblur.view.ActivityDetailsAdapter.getView(ActivityDetailsAdapter.java:71)
	at android.widget.HeaderViewListAdapter.getView(HeaderViewListAdapter.java:220)
	at android.widget.AbsListView.obtainView(AbsListView.java:2388)
	at android.widget.ListView.makeAndAddView(ListView.java:2052)
	at android.widget.ListView.fillDown(ListView.java:786)
	at android.widget.ListView.fillFromTop(ListView.java:847)
	at android.widget.ListView.layoutChildren(ListView.java:1798)
	at android.widget.AbsListView.onLayout(AbsListView.java:2187)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at androidx.viewpager.widget.ViewPager.onLayout(ViewPager.java:1775)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1083)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:148)
	at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:43)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:918)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1791)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1635)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1544)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
	at com.android.internal.policy.DecorView.onLayout(DecorView.java:761)
	at android.view.View.layout(View.java:19663)
	at android.view.ViewGroup.layout(ViewGroup.java:6075)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2500)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2216)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1396)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6773)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:966)
	at android.view.Choreographer.doCallbacks(Choreographer.java:778)
	at android.view.Choreographer.doFrame(Choreographer.java:713)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:952)
	at android.os.Handler.handleCallback(Handler.java:790)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6518)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

App version : **11.3.0 (202)**

**Comment by samuelclay** [comment]
Thanks for getting to this before me @shuvashish76. Another user emailed it in. 

> On my Pixel 6 Pro, the app doesn't crash completely when I try to open my profile. But it does try to load the lower part of the screen for around half a second before abruptly falling back to the feeds list.

**Comment by shuvashish76** [comment]
I assume this crash happened because of our new feature #1646 implementation...🤔 _(just a wild guess)_

---

[Also I noticed currently 1.Profile (top_left) & 2.Navigation Activities button (top_right) both does the same job, which seems unnecessary. I'll create an issue for this.]

**Comment by samuelclay** [comment]
Maybe, maybe not on #1646. It's a new activity type (`opml_import` and `opml_export`, look for it along with `feedsub` and others), but it should just be blank or show only the `content` if a new uncognized type. The type is really just for correct icons.

**Comment by sictiru** [comment]
I can replicate the crash with the latest build from PlayStore but however I don't get the same crash with a build off of  `master` or `sictiru`. Something is different between the builds but the place where the crash happens wasn't touched in years :thinking: https://github.com/samuelclay/NewsBlur/blob/8c56c6f98e767bc93d31c1c3f850bf5551b3f079/clients/android/NewsBlur/src/com/newsblur/view/ActivitiesAdapter.java#L101
@samuelclay Did something change for `/social/interactions` or `/social/activities` or `/social/profile` recently?

**Comment by samuelclay** [comment]
> @samuelclay Did something change for `/social/interactions` or `/social/activities` or `/social/profile` recently?

Yep, I added the `opml_import` and `opml_export` activities. Try both of those and you might be able to reproduce it.

**Comment by Netpilgrim** [comment]
Not sure if this info helps but I can also reproduce this crash with the same steps.
App version: 11.3.0
Android version: 12
Device: Samsung Galaxy S10

After the first crash I got this system message: “NewsBlur closed because this app has a bug. Try clearing the app's cache first, and then reopen the app.” On later crashes I got: “NewsBlur closed because this app has a bug. Try updating this app after its developer provides a fix for this error.”

**Comment by sictiru** [comment]
@Netpilgrim Appreciate the report! Version 11.3.1 is in the beta channel and it will be released very soon into production. That latest version should address the issue that you're encountering. If interested, you can join the public beta channel via Play Store.

**Comment by isomeme** [comment]
I am still seeing the crash when opening my account profile on Android.

**Comment by samuelclay** [comment]
Another user reported this by email, username smarkwell

**Comment by smarkwell** [comment]
Running 11.3.1 according to app info, (don't see a way to get version info inside application).  Still able to trigger crash in the same way.

**Comment by sictiru** [comment]
The changes on `/social/*` were more wide spread. Added additional checks based on the available logs.

**Comment by sictiru** [comment]
Fixed in NewsBlur v12.0

**Comment by smarkwell** [comment]
Confirmed fixed for me.  
Version 12.0
Android Version 13
Pixel 4a
